### ISSUE TITLE
On upgrade, with reuse-values, YAML parser errors looking for "document start"

### ISSUE THREAD

**Issue by nicdoye** [helm/helm#4336]
Output of `helm version`:
```
Client: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
```

Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.5", GitCommit:"32ac1c9073b132b8ba18aa830f46b77dcceb0723", GitTreeState:"clean", BuildDate:"2018-06-21T11:46:00Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.5", GitCommit:"32ac1c9073b132b8ba18aa830f46b77dcceb0723", GitTreeState:"clean", BuildDate:"2018-06-21T11:34:22Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
kops 1.9.1 on AWS

The YAML parser spits out an error claiming a `DOCUMENT-START` (`---`) is missing. I placed that at the beginning of `Chart.yaml`, `deployment.yaml` and `values.yaml`, but it still emits that error.

Example: https://github.com/nicdoye/helm-set-vs-reuse/tree/document-start-error/helm/parent/charts/subchart
(Note this is the `subchart` chart in `document-start-error` tag).

```console
$ helm install helm/parent/charts/subchart -n ds-error
NAME:   ds-error
...
$ helm upgrade ds-error helm/parent/charts/subchart  --reuse-values
Error: UPGRADE FAILED: yaml: line 2: did not find expected <document start>
```

**Comment by bryanlarsen** [comment]
@nicdoye, any resolution?   I'm seeing this too.

**Comment by nicdoye** [comment]
No, not yet. I'm waiting to see how @michelleN 's fix for https://github.com/helm/helm/issues/4336 (in https://github.com/helm/helm/pull/4364 ) affects things.

**Comment by nicdoye** [comment]
This is now resolved as of helm 2.10.0-rc3 .
